src/Microsoft.ApiDesignGuidelines.Analyzers/Core/UseIntegralOrStringArgumentForIndexers.cs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Analyzer.Utilities;
+
+namespace Microsoft.ApiDesignGuidelines.Analyzers
+{
+    /// <summary>
+    /// CA1043: Use Integral Or String Argument For Indexers
+    /// </summary>
+    public abstract class UseIntegralOrStringArgumentForIndexersAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string RuleId = "CA1043";
+
+        private static readonly LocalizableString s_localizableTitle = new LocalizableResourceString(nameof(MicrosoftApiDesignGuidelinesAnalyzersResources.UseIntegralOrStringArgumentForIndexersTitle), MicrosoftApiDesignGuidelinesAnalyzersResources.ResourceManager, typeof(MicrosoftApiDesignGuidelinesAnalyzersResources));
+
+        private static readonly LocalizableString s_localizableMessage = new LocalizableResourceString(nameof(MicrosoftApiDesignGuidelinesAnalyzersResources.UseIntegralOrStringArgumentForIndexersMessage), MicrosoftApiDesignGuidelinesAnalyzersResources.ResourceManager, typeof(MicrosoftApiDesignGuidelinesAnalyzersResources));
+        private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftApiDesignGuidelinesAnalyzersResources.UseIntegralOrStringArgumentForIndexersDescription), MicrosoftApiDesignGuidelinesAnalyzersResources.ResourceManager, typeof(MicrosoftApiDesignGuidelinesAnalyzersResources));
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(RuleId,
+                                                                             s_localizableTitle,
+                                                                             s_localizableMessage,
+                                                                             DiagnosticCategory.Performance,
+                                                                             DiagnosticSeverity.Warning,
+                                                                             isEnabledByDefault: false,
```

Comment:
Not very sure should be ON or OFF by default. The framework is not automatically generated, so think off is more safe? 
Should be on by default?